services/stripe_email_updater.go

```diff
@@ -0,0 +1,92 @@
+package services
+
+import (
+	"bytes"
+	"eth2-exporter/db"
+	"eth2-exporter/utils"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+)
+
+func stripeEmailUpdater(wg *sync.WaitGroup) {
+	firstRun := true
+	for {
+		// fetch all users with pending stripe email updates
+		var pendingUsers []struct {
+			Email            string `db:"email"`
+			StripeCustomerId string `db:"stripe_customer_id"`
+		}
+		err := db.FrontendReaderDB.Select(&pendingUsers, "SELECT email, COALESCE(stripe_customer_id. '') FROM users WHERE stripe_email_pending")
+		if err != nil {
+			utils.LogError(err, "error getting pending users for stripe email update service", 0)
+			time.Sleep(time.Second * 10)
+			continue
+		}
+
+		// update stripe customer email
+		var updatedUsers []string
+		for _, user := range pendingUsers {
+			if user.StripeCustomerId == "" {
+				utils.LogError(fmt.Errorf("user has no stripe_customer_id"), "error updating stripe customer email, this should never happen", 0, map[string]interface{}{"email": user.Email})
+				continue
+			}
+			err := updateStripeCustomerEmail(user.StripeCustomerId, user.Email)
```

Comment:
Fortunately stripe has very generous rate limits https://stripe.com/docs/rate-limits
But just in case that sometime we exceed 100 updates per second, we should add a short sleep in between requests just to make sure in case they change it sometime and as a safety so we don't bombard them in case of an error. Maybe something like 5 requests per second on our end

Reply:
fixed, ptal